### PR TITLE
[infra-proxy-service] API to update node attributes

### PR DESCRIPTION
**Pull request by vsingh-msys**
### :nut_and_bolt: Description: What code changed, and why?

- Infra service API to manage the node normal node attributes. 

### :chains: Related Resources
Fixes: #4532 

### :+1: Definition of Done

- Infra service, now able to update node normal attributes.

### :athletic_shoe: How to Build and Test the Change
- `rebuild components/infra-proxy-service`
- `rebuild components/automate-gateway`

Steps to test:

- To add data using 3rd internal steps https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views

- API to update node attributes:

```
curl -sSkH "api-token: $(get_admin_token)" -X PUT https://a2-dev.test/api/v0/infra/servers/chef-server-dev-test/orgs/chef-org-dev/nodes/aix-7.1/attributes?pretty -d '{"attributes": { "attr1": "abc"}}'

{
  "name": "aix-7.1",
  "attributes": "{\"attr1\":\"value1\"}"
}

```

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: components/infra-proxy-service/server/nodes.go [hunk 1]
```diff
@@ -144,7 +144,7 @@ func (s *Server) DeleteNode(ctx context.Context, req *request.DeleteNode) (*resp
 	}, nil
 }
 
-// UpdateNode update the node attributes
+// UpdateNode update the node
```
Comment (msorens) [inline components/infra-proxy-service/server/nodes.go:147]:
```suggestion
// UpdateNode updates the node
```

File: components/infra-proxy-service/server/nodes.go [hunk 2]
```diff
@@ -191,6 +191,51 @@ func (s *Server) UpdateNode(ctx context.Context, req *request.UpdateNode) (*resp
 	}, nil
 }
 
+// UpdateNodeAttributes update the node attributes
```
Comment (msorens) [inline components/infra-proxy-service/server/nodes.go:194]:
```suggestion
// UpdateNodeAttributes updates the node attributes
```

File: components/infra-proxy-service/server/nodes.go [hunk 3]
```diff
@@ -191,6 +191,51 @@ func (s *Server) UpdateNode(ctx context.Context, req *request.UpdateNode) (*resp
...
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &response.UpdateNodeAttributes{
+		Name:       req.Name,
+		Attributes: string(resAttributes),
+	}, nil
+}
+
 // fetchAffectedNodes get the nodes used by chef object
```
Comment (msorens) [inline components/infra-proxy-service/server/nodes.go:239]:
```suggestion
// fetchAffectedNodes gets the nodes used by chef objects
```

**Review by punitmundra — CHANGES_REQUESTED** [review]
not able to test as per given instructions.

**Comment by vsingh-msys** [comment]
> not able to test as per given instructions.

@punitmundra could you please elaborate it more.

I followed the same steps to test it out

- git checkout/fetch VSingh/infra-proxy-service-node-update-attribute
- running hab command if not already
- rebuild components/automate-gateway
- rebuild components/automate-gateway
- curl -sSkH "api-token: $(get_admin_token)" -X PUT https://a2-dev.test/api/v0/infra/servers/chef-server-dev-test/orgs/chef-org-dev/nodes/aix-7.1/attributes?pretty -d '{"attributes": { "attr1": "abc"}}'


Providing error messages or command running would be helpful?

**Comment by punitmundra** [comment]
> > not able to test as per given instructions.
> 
> @punitmundra could you please elaborate it more.
> 
> I followed the same steps to test it out
> 
> * git checkout/fetch VSingh/infra-proxy-service-node-update-attribute
> * running hab command if not already
> * rebuild components/automate-gateway
> * rebuild components/infra-proxy-service
> * curl -sSkH "api-token: $(get_admin_token)" -X PUT https://a2-dev.test/api/v0/infra/servers/chef-server-dev-test/orgs/chef-org-dev/nodes/aix-7.1/attributes?pretty -d '{"attributes": { "attr1": "abc"}}'
> 
> Providing error messages or command running would be helpful?

@vsingh-msys : how to add the node, that step i am not clear.

**Comment by vsingh-msys** [comment]
@punitmundra Adding node directly from API is not in scope, nodes getting added into chef-server when a node gets bootstraps. Some of the updates operations allowed like node environment, node attributes, node run list, and reset node key.

**Review by punitmundra — APPROVED** [review]
change are fine. i have tested. code looks fine to me.

**Review by kalroy — APPROVED** [review]

**Pull request merged**